test/correctness/simd_op_check.cpp

```diff
@@ -1327,6 +1327,13 @@ void check_hvx_all() {
         hvx_width = 128;
     }
 
+    // Verify that unaligned loads use the right instructions, and don't try to use
+    // immediates of more than 3 bits.
+    check("valign(v*,v*,#7)", hvx_width/1, in_u8(x + 7));
+    check("vlalign(v*,v*,#7)", hvx_width/1, in_u8(x + hvx_width - 7));
+    check("valign(v*,v*,r*)", hvx_width/1, in_u8(x + 8));
+    check("valign(v*,v*,r*)", hvx_width/1, in_u8(x + hvx_width - 8));
+
```

Comment:
It might be useful to test for element_size > 8 bits. This is because the scalar arg of valign is in bytes. So if we are off by 2 elements of 16 bits, then that arg should be 4. This will test if we are moving from elements to bytes correctly when we generate valigns from shuffle_vectors.